homeassistant/config_entries.py

```diff
@@ -819,6 +817,9 @@ async def async_unload(
                     hass, ConfigEntryState.FAILED_UNLOAD, str(exc) or "Unknown error"
                 )
             return False
+        finally:
+            if domain_is_integration and hasattr(self, "runtime_data"):
```

Comment:
Actually reading the code now better, we already call `async_unload_entry`, which is where teardown happens, before we delete the runtime data.

I am not sure what this PR adds?

Reply:
`_async_process_on_unload` (line 810) is where we call the things added with `ConfigEntry.async_on_unload`